tests/tools/fswalker/protofile/protofile.go

```diff
@@ -3,17 +3,23 @@
 package protofile
 
 import (
+	"bytes"
 	"io/ioutil"
-	"strings"
 
-	"github.com/golang/protobuf/proto"
+	"google.golang.org/protobuf/encoding/prototext"
+	"google.golang.org/protobuf/proto"
 )
 
 // WriteTextProto writes a text format proto buf for the provided proto message.
 func WriteTextProto(path string, pb proto.Message) error {
-	blob := proto.MarshalTextString(pb)
+	blob, err := prototext.Marshal(pb)
```

Comment:
@redgoat650 Does this work for you or does it break anything?
There is also [`prototext.Format`](https://pkg.go.dev/google.golang.org/protobuf/encoding/prototext?tab=doc#Format), but not sure whether that works.

Reply:
So the reason I'm using `"github.com/golang/protobuf/proto"` here was because that's the library fswalker is using to read the file.
The easiest way I found to invoke the exported fswalker packages was to write out the protobuf config file and let the library open and decode it. Assuming the new protobuf encoding package is compatible with the package used in fswalker this should work, but it may be safer to just use the same proto package.

Reply:
For the encoding, I'm assuming it should work with the current change, as in I don't expect changes in the text format, but I could be wrong. I did not read through all the notes about incompatible changes in the release logs.

Currently this change is failing to compile due to a type mismatch. I have not tracked down the issue yet.

IIUC, the _"superseded "_ `github.com/golang/protobuf/proto` is not built atop `google.golang.org/protobuf` in order to preserve the API, but the underlying implementation is the same. Anyway, need to dig further because this is now preventing dependency upgrades even when protobuf is not upgraded. A golangci linter flags the import as deprecated. I am not sure why it does not fail for other PRs though.

Reply:
Yeah you're right; the protobuf spec itself didn't change, and the config files are the simplest of protobuf structures. No reason to believe the "superseded" reader wouldn't be able to unmarshal from a file created by the new library. And the unit tests should catch any obvious issues (`walker_test.go`, and `fswalker_test.go`)